app/src/main/java/com/itsaky/androidide/activities/editor/EditorHandlerActivity.kt

```diff
@@ -457,7 +467,12 @@ open class EditorHandlerActivity : ProjectHandlerActivity(), IEditorHandler {
     val finalModified = modified
 
     withContext(Dispatchers.Main) {
-      editorViewModel.setFilesModified(finalModified)
+      editorViewModel.apply {
+        setFilesModified(finalModified)
+        if (updateSaving) {
+          setFilesSaving(false)
+        }
```

Comment:
The saving flag is only reset if the file has been successfully saved. You also need to reset the state if the save fails :

```kotlin
if (!frag.save()) {
    if (updateSaving) {
        setFilesSaving(false)
    }
    return
}
```